### PR TITLE
BaSP-MR-33: SUPERADMIN TEST - Get, Get by Id & Create

### PR DESCRIPTION
**Pull request by keiler-art**

### CODE REVIEW

File: src/tests/super-admins.test.js [hunk 1]
```diff
@@ -1,5 +1,88 @@
...
+describe('get/api/super-admins/:id', () => {
+  test('should return status 200', async () => {
+    const response = await request(app).get('/api/super-admins/635316fe464e1ad6227622e4').send();
+    expect(response.status).toBe(200);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.data).toBeDefined();
+    expect(response.body.message).toBe('Superadmin found successfully!');
+  });
+  test('should return status 404 when Id is not found', async () => {
+    const response = await request(app).get('/api/super-admins/635316fe464e1ad6227622e5').send();
```
Comment (JuanManuel-Blois) [inline src/tests/super-admins.test.js:52]: Create a variable called InvalidSuperAdminMockup with this invalid ID to check it

File: src/tests/super-admins.test.js [hunk 2]
```diff
@@ -1,5 +1,88 @@
...
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.data.length).toBe(undefined);
+    expect(response.body.message).toBeDefined();
+  });
+  test('should return status 500', async () => {
+    jest.spyOn(SuperAdmin, 'findById').mockRejectedValue(new Error('Something went wrong'));
+    const response = await request(app).get('/api/super-admins/635316fe464e1ad6227622e4').send();
```
Comment (JuanManuel-Blois) [inline src/tests/super-admins.test.js:60]: replace this Id with the variable where you call the seed id

Comment (keiler-art) [reply src/tests/super-admins.test.js]: Done!

File: src/tests/super-admins.test.js [hunk 3]
```diff
@@ -1,5 +1,96 @@
...
+import superAdminSeed from '../seeds/super-admins';
+
+const mockSuper = {
+  email: 'superadmin@super.com',
+  password: 'superadmin123',
+};
+
+const mockWrongId = {
+  id: '635316fe464e1ad6227622e5',
+};
```
Comment (nicolasblasco) [inline src/tests/super-admins.test.js:13]: I think we shouldn't use an object for the id. We could directly do this `const mockWrongId = '635316fe464e1ad6227622e5'`

Comment (nicolasblasco) [inline src/tests/super-admins.test.js:13]: I think we shouldn't use an object for the id. We could directly do this `const mockWrongId = '635316fe464e1ad6227622e5'`

File: src/tests/super-admins.test.js [hunk 4]
```diff
@@ -1,5 +1,96 @@
...
+afterEach(async () => {
+  await SuperAdmin.collection.deleteMany();
+});
+
+describe('get/api/super-admins', () => {
+  test('should return status 200', async () => {
+    const response = await request(app).get('/api/super-admins').send();
+    expect(response.status).toBe(200);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.data).toBeDefined();
```
Comment (nicolasblasco) [inline src/tests/super-admins.test.js:32]: We could test the data.length `expect(response.body.data.length).toBeGreaterThan(0)`

File: src/tests/super-admins.test.js [hunk 5]
```diff
@@ -1,5 +1,96 @@
...
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.data.length).toBe(0);
+    expect(response.body.message).toBe('No superadmins found!');
+  });
+  test('should return status 500', async () => {
+    jest.spyOn(SuperAdmin, 'find').mockRejectedValue(new Error('Something went wrong'));
+    const response = await request(app).get('/api/super-admins').send();
+    expect(response.status).toBe(500);
+    expect(response.error).toBeTruthy();
```
Comment (nicolasblasco) [inline src/tests/super-admins.test.js:47]: We should test the message, data and error of the body of the response

File: src/tests/super-admins.test.js [hunk 6]
```diff
@@ -1,5 +1,96 @@
...
+    expect(response.status).toBe(200);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.data).toBeDefined();
+    expect(response.body.message).toBe('Superadmin found successfully!');
+  });
+  test('should return status 404 when Id is not found', async () => {
+    const response = await request(app).get(`/api/super-admins/${mockWrongId.id}`).send();
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.data.length).toBe(undefined);
```
Comment (nicolasblasco) [inline src/tests/super-admins.test.js:63]: Are we sure that we should test the length of an object? I think we could test the data like this `expect(response.body.data).toBeEqual({})`

File: src/tests/super-admins.test.js [hunk 7]
```diff
@@ -1,5 +1,96 @@
...
+    expect(response.body.message).toBeDefined();
+  });
+  test('should return status 500', async () => {
+    jest.spyOn(SuperAdmin, 'findById').mockRejectedValue(new Error('Something went wrong'));
+    const response = await request(app).get(`/api/super-admins/${mockCoolId.id}`).send();
+    expect(response.status).toBe(500);
+    expect(response.error).toBeTruthy();
+  });
```
Comment (nicolasblasco) [inline src/tests/super-admins.test.js:71]: We should test the message, data and error of the body of the response

File: src/tests/super-admins.test.js [hunk 8]
```diff
@@ -1,5 +1,96 @@
...
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.data).toBeDefined();
+    expect(response.body.message).toBe('SuperAdmin could not be found and created!');
+  });
+  test('should return status 500', async () => {
+    jest.spyOn(SuperAdmin, 'create').mockRejectedValue(new Error('Oops something went wrong'));
+    const response = await request(app).post('/api/super-admins/').send(mockSuper);
+    expect(response.status).toBe(500);
+    expect(response.error).toBeTruthy();
```
Comment (nicolasblasco) [inline src/tests/super-admins.test.js:94]: We should test the message, data and error of the body of the response

**Review by nicolasblasco — CHANGES_REQUESTED** [review]
Please add a description to our pull request

**Pull request merged**